### ISSUE TITLE
Suspicious behaviour related to Transitions

### ISSUE THREAD

**Issue by alcinos** [mltframework/mlt#225]
Hi there,

I was observing some incoherent behaviours with transitions, so I started digging around to find an explanation to this.
The first thing I noticed is that for some reason, mlt_field_plant_transition was always returning 0, regardless of the actual outcome of the command (see https://github.com/mltframework/mlt/pull/224)

But after fixing that, things turned even more strange. The toy example I ended up with is the following: add a transition, disconnect it, and add it back again:
```
  Factory::init();
  Profile profile;

  Playlist track1(profile), track2(profile);

  Transition transition(profile, "composite");

  Tractor tractor;
  tractor.insert_track(track1, 0);
  tractor.insert_track(track2, 1);

  Field *field = tractor.field();
  int err = field->plant_transition(transition, 1, 0);
  std::cout<<err<<std::endl;

  field->disconnect_service(transition);

  err = field->plant_transition(transition, 1, 0);
  std::cout<<err<<std::endl;
```
The expected output of this is 0\n0, but with the patch proposed in the above PR, I get 0\n3. And the transition is not correctly added.
My analysis is the following: the disconnect_service function, aka mlt_field_disconnect_service, only removes the producer from the producer chain. But I think that's not enough, because the ```in``` array of the transition service doesn't get cleaned uppon deletion, hence when reinserted, it is still full of the previous data left by previous insertion. That's why on the second call, mlt_service_connect_producer (which is called by insert_track) returns 3 instead of 0, breaking the subsequent addition of the transition (in particular, it is not added to the producer chain).

Any thoughts on this ?

**Comment by ddennedy** [comment]
Your analysis is correct, the transition was merely taken out of the chain, but the transition still refers to the producer. Perhaps to do what you want, you can call mlt_service_disconnect_producer() on the transition.

**Comment by alcinos** [comment]
Thanks for your answer. So adding the line 
```
    field->disconnect_service(transition);
    transition.disconnect_producer(transition.get_a_track());
```
leads to the correct behaviour.
I wonder if this call shouldn't be made by the field_disconnect_service function ?
Or at least that should be documented somehow. In particular, I don't think that the fact that the producer is inserted with index=a_track is documented anywhere (I had to infer it from the code)

**Comment by alcinos** [comment]
Well actually my proposed fix is not entirely correct...
On insertion, the producer is indeed inserted with index=a_track, but nothing prevents the a_track of the transition to be modified afterwards, and AFAIK, there is no way to retrieve the "original" a_track...

**Comment by ddennedy** [comment]
OK, well, you are going to have to work with the way it is for the most because I am not going to change things much.

**Comment by alcinos** [comment]
I see. If I PR a function "mlt_service_disconnect_all_producers", would you accept it ? I can't see any way of implementing it from the outside since there is no access to the "in" array of the service.

**Comment by ddennedy** [comment]
yes

**Comment by ddennedy** [comment]
Resolved by #226
